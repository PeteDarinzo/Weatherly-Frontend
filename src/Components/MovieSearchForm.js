import React, { useState } from "react";


const MovieSearchForm = ({ getMovies }) => {

  const intialState = {
    title: ""
  }

  const [formData, setFormData] = useState(intialState);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData(formData => ({
      ...formData,
      [name]: value
    }));
  }


  const handleSubmit = (e) => {
    e.preventDefault();
    getMovies(formData.title);
  }

  return (
    <div>
      <form onSubmit={handleSubmit}>
        <label htmlFor="title">Title</label>
        <input
          id="title"
          type="text"
          name="title"
          value={formData.title}
          onChange={handleChange}
          placeholder="Enter title" />
        <button>Search</button>
      </form>
    </div>
  );
}

export default MovieSearchForm;