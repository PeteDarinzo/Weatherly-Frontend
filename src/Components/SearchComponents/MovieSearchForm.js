import React, { useState } from "react";
import Container from '@mui/material/Container';
import TextField from '@mui/material/TextField';
import Typography from '@mui/material/Typography';
import Button from '@mui/material/Button';
import FormControl from '@mui/material/FormControl';
import Box from '@mui/material/Box';
import { makeStyles } from "@mui/styles";

const useStyles = makeStyles({
  form: {
    margin: 20,
    padding: 20
  },
  field: {
    marginTop: 20,
    marginBottom: 20,
    display: "block"
  }
});

const MovieSearchForm = ({ getMovies }) => {

  const classes = useStyles();

  const intialState = {
    title: ""
  }

  const [formData, setFormData] = useState(intialState);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData(formData => ({
      ...formData,
      [name]: value
    }));
  }


  const handleSubmit = (e) => {
    e.preventDefault();
    getMovies(formData.title);
  }

  return (
    <Container>
      <Box className={classes.form}>
        <Typography variant="h4">Search Titles</Typography>
        <form onSubmit={handleSubmit}>
          <FormControl fullWidth>
            <TextField
              className={classes.field}
              name="title"
              id="title"
              label="Title"
              variant="outlined"
              value={formData.title}
              fullWidth
              onChange={handleChange}
              required />
            <Button
              type="submit"
              variant="contained"
              color="primary"
              size="large">Search</Button>
          </FormControl>
        </form>
      </Box>
    </Container>
  );
}

export default MovieSearchForm;