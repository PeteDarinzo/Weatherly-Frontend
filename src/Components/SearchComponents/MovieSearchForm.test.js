import MovieSearchForm from "./MovieSearchForm";
import { render } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom";

it("renders a search form without crashing", () => {
  render(
    <MemoryRouter>
      <MovieSearchForm getMovies={() => { }} />
    </MemoryRouter>
  );
});


it("matches the search form snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <MovieSearchForm getMovies={() => { }} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
