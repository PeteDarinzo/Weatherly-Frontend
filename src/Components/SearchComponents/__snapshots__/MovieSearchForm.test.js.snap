// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the search form snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="makeStyles-form-3 MuiBox-root css-0"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
      >
        Search Titles
      </h4>
      <form>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root makeStyles-field-4 css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="title"
              id="title-label"
            >
              Title
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="title"
                name="title"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1g8cwq9"
                >
                  <span>
                    Title *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-jh47zj-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            Search
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
